core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -212,7 +212,7 @@ final class KafkaMetadataLog private (
   }
 
   override def flush(): Unit = {
-    log.flush()
+    log.flush(true)
```

Comment:
This is called on every log append. We only need to do the inclusive part in close().